Improve toggle button for disabled combo boxes

AlexGustafsson opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue or discussion
  in the additional information section.
-->

Whenever a combo box is disabled, hide the open/close icon.

Below are some screenshots, with the example UI being from #24736.

Before:

Idle:

![image](https://github.com/user-attachments/assets/ce0bfbf1-0d8f-4036-9407-d04d10917cec)

When clicked:

![image](https://github.com/user-attachments/assets/31c0c598-9bda-487b-a1a1-a92fb5d47d19)

After:

![image](https://github.com/user-attachments/assets/f0acec11-bf10-42fd-b616-610e2dc72be9)

## Type of change
<!--
  What type of change does your PR introduce to the Home Assistant frontend?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR.
-->

N/A

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

N/A

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

[docs-repository]: https://github.com/home-assistant/home-assistant.io

AlexGustafsson commented:
If it's important to still convey that it's a "dropdown" when disabled, I propose that we "grey out" the icon when disabled and remove hover style and click events instead.

matthiasdebaat commented:
Material Design got the following [guidelines](https://m3.material.io/components/text-fields/specs#544d001a-e0ce-49eb-b77b-89fe08acccc5)
<img width="1024" alt="unnamed" src="https://github.com/user-attachments/assets/a55293e6-604c-4102-98fd-538947872449" />

Lets fix the combo box by:
- Change the "arrow down" icon to the same color/transparency as the label and input.
- Don't make the "arrow down" icon clickable.

AlexGustafsson committed:
Improve toggle button for disabled combo boxes

- Disable pointer events for the toggle button when the combo box is
  disabled
- Align the color of the toggle button with other controls when the
  combo box is disabled

AlexGustafsson commented:
@matthiasdebaat Thanks for taking a look. Fixed. The color change isn't that noticeable, but it is the same color as the label / text now.

<img width="634" alt="image" src="https://github.com/user-attachments/assets/deca5d20-6c8a-4cd8-93b3-cac823000a48" />

AlexGustafsson commented on src/components/ha-combo-box.ts:
```diff
@@ -359,6 +360,10 @@ export class HaComboBox extends LitElement {
       color: var(--primary-color);
     }
+    .toggle-button[disabled] {
+      color: var(--mdc-text-field-disabled-ink-color, rgba(0, 0, 0, 0.38));
```
Fixed.
